File: gex/src/model/StartRegionData.ts

export class StartRegionData {

    public allyTeamID: number = 0;
    public regions: StartRegion[] = [];

    public static parse(elem: any): StartRegionData {
        return {
            allyTeamID: elem.allyTeamID,
            regions: elem.regions.map((iter: any) => StartRegion.parse(iter))
        };
    }

}

export class StartRegion {

    public vertices: StartRegionVertex[] = [];

    public static parse(elem: any): StartRegion {
        return {
            vertices: elem.vertices.map((iter: any) => StartRegionVertex.parse(iter))
        };
    }

}

export class StartRegionVertex {
    public x: number = 0;
    public z: number = 0;

    public static parse(elem: any): StartRegionVertex {
        return {
            x: elem.x,
            z: elem.z
        };
    }

}

File: gex.Common/Models/Map/StartRegionData.cs
using gex.Common.Services.Util;
using System.Collections.Generic;

namespace gex.Common.Models.Map {

    public class StartRegionData {

        public int AllyTeamID { get; set; }

        public List<StartRegion> Regions { get; set; } = [];

    }

    public class StartRegion {

        public List<PolygonStartboxUtil.Pair> Vertices { get; set; } = [];

    }
}
